### PR TITLE
Add some bindings to vscode diagnostics manipulation

### PR DESCRIPTION
**Pull request by ulugbekna**
I had this commit as part of #640 , but since I'm not sure we're going further with the approach of #640, decided to create another PR. 

The bindings aren't full, but were somewhat tricky to get right (at least for me as a newbie to ojs). 

Fixed a "deprecated" warning along the way.

### CODE REVIEW

**Comment by mnxn** [comment]
I added support for the `relatedInformation` and `tags` fields as well as properly handling the record type for `code`.

**Review by mnxn — APPROVED** [review]

File: vscode/vscode/vscode.ml [hunk 1]
```diff
@@ -1182,6 +1182,99 @@ module ProgressOptions = struct
...
+     | `Int of int
+     | `Targeted of code_target
+     ]
+    [@js.union])
+  [@@js]
+
+  let code_of_js js_val =
+    match Ojs.type_of js_val with
+    | "object" -> `Targeted ([%js.to: code_target] js_val)
+    | _ -> string_or_int_of_js js_val
```
Comment (ulugbekna) [inline vscode/vscode/vscode.ml:1247]: doesn't it make sense to inline `string_or_int_of_js` unless we want to put it in a larger scope to be reusable?

**Comment by ulugbekna** [comment]
btw, feel free to merge when you consider it ready

File: vscode/vscode/vscode.ml (hunk 1 above)
Comment (mnxn) [reply vscode/vscode/vscode.ml:1247]: `string_or_int_of_js` is being used (by gen_js_api) for the `value` field of `code_target`.
So the function is already being reused locally.

Comment (mnxn) [reply vscode/vscode/vscode.ml:1247]: We could add these kind of types to the `interop` module, but I don't see the need for it yet.

**Pull request merged**